File: api/ServerManager.Rest/Dto/CreateServerRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ServerManager.Rest.Dto
{
    public class CreateServerRequest
    {
        public string Name { get; set; }
        public string Version { get; set; }
        public string Description { get; set; }
        public ServerPropertyList Properties { get; set; }
    }
}

File: ui/management-console/src/models/CreateServerRequest.ts
import { ServerProperty } from "./ServerProperty";

export class CreateServerRequest {
	public constructor() {
		this.name = "";
		this.version = "";
		this.description = "";
		this.properties = [];
	}
	
	name: string;
	version: string;
	description: string;
	properties: ServerProperty[];
}
